rocketpy/simulation/df2flight.py

```diff
@@ -0,0 +1,208 @@
+"""Starts with a pandas DataFrame containing the rocket's collected flight data
+and build a rocketpy.Flight object from it.
+"""
+
+from dataclasses import dataclass
+from functools import cached_property, wraps
+
+import numpy as np
+
+from rocketpy.mathutils import Function
+from rocketpy.tools import import_optional_dependency
+
+pandas = import_optional_dependency("pandas")
+pd = pandas
+
+
+def df2property(
+    col_name, output_name, interpolation="spline", extrapolation="constant"
+):
+    """Decorator to create a property from a data frame. The property works as a
+    rocketpy.Function object. Check the DataFrameToFlight class for additional
+    details
+
+    Parameters
+    ----------
+    col_name : str
+        The name of the column in the data frame. This is case sensitive.
+    output_name : str
+        The label of the output to be used in the rocketpy.Function object.
+    interpolation : str, optional
+        The interpolation method to be used in the rocketpy.Function object.
+        The default is "spline".
+    extrapolation : str, optional
+        The extrapolation method to be used in the rocketpy.Function object.
+        The default is "constant".
+
+    Returns
+    -------
+    decorator
+        The decorator to be used in the DataFrameToFlight class. This only works
+        in the DataFrameToFlight class.
+    """
+
+    class Decorator:
+        """A decorator class that wraps a function and provides create a
+        property from a data frame. This is made via class because it's the only
+        way to use the __set_name__ method. This method is used to get the name
+        of the property in the class.
```

Comment:
```suggestion
        """A decorator class that wraps a method to create a property from a data
        frame, as a Function instance.
        This is made via class because it's the only way to use the __set_name__
        method. This method is used to get the name of the property in the class.
```